bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -144,6 +143,78 @@ public boolean columnExists(String tableName, String columnName, CaseDbTransacti
         return columnExists;
     }
 	
+	/**
+	 * Checks if a table exists in the case database.
+	 *
+	 * @param tableName name of the table to check
+	 * 
+	 * @return true if the table already exists, false otherwise
+	 * @throws TskCoreException 
+	 */
+	public boolean tableExists(String tableName) throws TskCoreException {
+		
+		boolean doesTableExist = false;
+		CaseDbTransaction localTrans = tskDB.beginTransaction();
+        try {
+			doesTableExist = tableExists(tableName, localTrans);
+			localTrans.commit();
+			localTrans = null;
+        } 
+		finally {
+			if (null != localTrans) {
+				try {
+					localTrans.rollback();
+				} catch (TskCoreException ex) {
+					logger.log(Level.SEVERE, "Failed to rollback transaction after exception", ex); //NON-NLS
+				}
+			}
+		}
+		
+        return doesTableExist;
+    }
+	
+	/**
+	 * Checks if a table exists in the case database.
+	 *
+	 * @param tableName name of the table to check
+	 * @param transaction transaction 
+	 * 
+	 * @return true if the table already exists, false otherwise
+	 * @throws TskCoreException 
+	 */
+	public boolean tableExists(String tableName, CaseDbTransaction transaction) throws TskCoreException {
+		
+		boolean tableExists = false;
+        Statement statement = null;
+        try {
+			CaseDbConnection connection = transaction.getConnection();
+			statement = connection.createStatement();
+			if (DbType.SQLITE == tskDB.getDatabaseType()) {
+				ResultSet tableQueryResults = statement.executeQuery("SELECT name FROM sqlite_master WHERE type='table'");  //NON-NLS
+				while (tableQueryResults.next()) {
+					if (tableQueryResults.getString("name").equalsIgnoreCase(tableName)) { //NON-NLS
+						tableExists = true;
```

Comment:
I think it would become a bit simpler logic if instead of querying for "SELECT name FROM sqlite_master WHERE type='table'"  we instead queried for ""SELECT name FROM sqlite_master WHERE type='table' and name = '%s'" with %s being the table name this would allow us to remove the looping through results and have the query be the only thing different about the SQLite and Postgres logic, as both would be checking if results exists to determine if the table was present.

But I might be missing something.